build/rpmfc.c

```diff
@@ -1235,9 +1235,16 @@ rpmRC rpmfcClassify(rpmfc fc, ARGV_t argv, rpm_mode_t * fmode)
 	case S_IFREG:
 	default:
 	    /* XXX all files with extension ".pm" are perl modules for now. */
-	    if (rpmFileHasSuffix(s, ".pm"))
+	    if (rpmFileHasSuffix(s, ".pm")) {
 		ftype = "Perl5 module source text";
-
+		fmime = "text/plain";
+	    } else if (rpmFileHasSuffix(s, ".c")) {
+		ftype = "C Code";
+		fmime = "text/x-c";
+	    } else if (rpmFileHasSuffix(s, ".h")) {
+		ftype = "C Header";
+		fmime = "text/x-c";
+            }
 	    /* XXX all files with extension ".la" are libtool for now. */
 	    else if (rpmFileHasSuffix(s, ".la"))
 		ftype = "libtool library file";
```

Comment:
missing `fmime` assignment?

Reply:
I didn't want to go to far, but first hear the feedback.